### ISSUE TITLE
Make Usable with PHP 8

### ISSUE THREAD

**Issue by mestrega** [vimeo/vimeo.php#334]
Library should be made usable with PHP 8.x

**Comment by aaronm67** [comment]
This library is usable with PHP 8.x, the only php 7.x requirements are in `require-dev` libraries.

**Comment by mestrega** [comment]
The issue is that this is not compatible with Drupal 9 on PHP 8 as it also pertains to ankitpokhrel/tus-php. Still in the process of getting all the requirements before adding more information to this. 

The issue is that D9 requires symfony/console 4.4, but the PHP8 compatible version of this requires a version of tus-php that requires symfony/console 5 and fails to build. There needs to be an intermediate step that allows for symfony/console 4.4 on PHP 8.

**Comment by mestrega** [comment]
I am currently working on an intermediary fork that would address both of these issues.

**Comment by aaronm67** [comment]
Sounds good -- I'll reopen pending details, but that sounds like it's a normal library incompatibility, not anything with this library or php8. 

If it's a symfony/console incompatibility, it's most likely related to Drupal 9 being EOLed and using outdated libraries, as tus-php uses [symfony/console 5|6](https://github.com/ankitpokhrel/tus-php/blob/main/composer.json#L19), so we wouldn't be able to merge any fixes in this regard.

**Comment by mestrega** [comment]
You are correct. This is really a temporary fix to bridge the gap between upgrading to PHP8 and then Drupal 10. It may not even need to be merged in, but could benifit from having a patch that could be accessed here and applied via a dev-* branch for others in similar circumstances. 
Thank you very much!

**Comment by mestrega** [comment]
For reference on Work in Progress, the Forked Branches being worked on are as follows:

For vimeo/vimeo.php: https://github.com/mestrega/vimeo.php/tree/PHP-8-Compat

For ankitpokhrel/tus-php: https://github.com/mestrega/tus-php/tree/Feature/Make-Compat-With-EOL-D9

**Comment by mestrega** [comment]
I attempted to create a PR to link my changes to this ticket, but am unable to link it to this ticket . Can I get guidance on that?

The desired branch to attach is https://github.com/mestrega/vimeo.php/tree/PHP-8-Compat

**Comment by aaronm67** [comment]
I think in the current state, this is probably best maintained as a fork - there's a significant amount of risk in pointing to a `dev-*` library.  If the tus-php maintainers will officially support a drupal 9 branch, we'd be happy to also support one.

[This](https://github.com/vimeo/vimeo.php/compare/master...mestrega:vimeo.php:PHP-8-Compat#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R21) is a unnecessary, `>=7.1.0` includes `^8.1.0`. this branch is a "Drupal 9" support branch, not a PHP 8 compatibility branch.
